internal/service/certificatesigningrequest.go

```diff
@@ -264,19 +261,33 @@ func (h *ServiceHandler) ApproveCertificateSigningRequest(ctx context.Context, r
 		if errors.Is(err, flterrors.ErrResourceNotFound) {
 			return server.ApproveCertificateSigningRequest404JSONResponse{}, nil
 		}
-		return nil, err
+		return server.ApproveCertificateSigningRequest500JSONResponse{Message: err.Error()}, nil
+	}
+	if csr == nil {
+		return server.ApproveCertificateSigningRequest500JSONResponse{Message: "CertificateSigningRequest is nil"}, nil
 	}
 
+	// do not approve a denied request, or recreate a cert for an already-approved request
 	if api.IsStatusConditionTrue(csr.Status.Conditions, api.CertificateSigningRequestDenied) {
 		return server.ApproveCertificateSigningRequest409JSONResponse{Message: "The request has already been denied"}, nil
 	}
 	if api.IsStatusConditionTrue(csr.Status.Conditions, api.CertificateSigningRequestApproved) {
 		return server.ApproveCertificateSigningRequest409JSONResponse{Message: "The request has already been approved"}, nil
 	}
 
-	signedCert, err := signApprovedCertificateSigningRequest(h.ca, csr)
+	signedCert, err := signApprovedCertificateSigningRequest(h.ca, *csr)
 	if err != nil {
-		return nil, err
+		switch {
+		case errors.Is(err, flterrors.ErrCNLength) ||
```

Comment:
I wonder if not better to create SignatureError and return it? and not to verify list of errors?

Reply:
The problem I encountered was that there can be many types of errors that occur during `signApprovedCertificateSigningRequest`, and generalizing over the 400 or 422 errors doesn't give as much information to the user as they sometimes need to print different specifics.

That being said - in a near-future PR I could make an error category for all of these so that this reads a little cleaner.